cocos/2d/CCActionInterval.cpp

```diff
@@ -141,6 +133,8 @@ void ActionInterval::step(float dt)
     if (sendUpdateEventToScript(updateDt, this)) return;
     
     this->update(updateDt);
+
+    _done = _elapsed >= _duration;
```

Comment:
If I do the test in `isDone` for an action with a zero duration then the action may say it is done while it is not. Consider the following case:

```c++
CallFunc* func( CallFunc::create( [] {} ) );
Action* action( Sequence::create( func, func, nullptr ) );
// the action was not started, it should not be done
assert( !action->isDone() );
```

The sequence is made of instant actions, thus its duration is zero. In this case `_elapsed >= _duration` is verified even if the action was not started. By using a flag updated when the action is updated we ensure that the action has been executed at least once before being done.